### PR TITLE
411_4_2 chronik sync transactions

### PR DESCRIPTION
**Pull request by vorg-code**
Description

Implements blockchain client chronik: sync transactions 

Test plan

1. Create new button with address (http://localhost:3000/buttons): ecash:qp0kfg0uk64k4pa7tw226djx4v30attzzslrec0ypu
2. Change ’grpc’ to ‘chronik’ for XEC in `constants/index.ts`, like this:
export const NETWORK_BLOCKCHAIN_CLIENTS: KeyValueT<BLOCKCHAIN_CLIENT_OPTIONS> = {
  ecash: `'chronik'`,
  bitcoincash: 'grpc'
}
3. Visit: http://localhost:3000/api/address/transactions/sync/ecash:qp0kfg0uk64k4pa7tw226djx4v30attzzslrec0ypu

This will use chronik to sync this address transactions and you will see 5 new transactions if you query the database or open the button you created in step 1.

### CODE REVIEW

File: services/chronikService.ts [hunk 1]
```diff
@@ -59,73 +143,73 @@ export class ChronikBlockchainClient implements BlockchainClient {
...
-    const buffer = Buffer.from(hash160, 'hex')
+  }
 
-    // Because ecashaddrjs only accepts Uint8Array as input type, convert
-    const hash160ArrayBuffer = new ArrayBuffer(buffer.length)
-    const hash160Uint8Array = new Uint8Array(hash160ArrayBuffer)
-    for (let i = 0; i < hash160Uint8Array.length; i += 1) {
-      hash160Uint8Array[i] = buffer[i]
-    }
+  // Test hash160 for correct length
```
Comment (Klakurka) [inline services/chronikService.ts:193]: No need for comments that are just English versions of the code. Better to explain why we're bothering to check it.

**Review by Klakurka — CHANGES_REQUESTED** [review]
Test plan works for me but I'm not sure how to sanity check that it's actually using Chronik when I change it over in the constants file.

**Comment by vorg-code** [comment]
@Klakurka you can place a `console.log()` inside the chronik method

File: services/chronikService.ts (hunk 1 above)
Comment (vorg-code) [reply services/chronikService.ts:193]: obvious comments removed

**Comment by Klakurka** [comment]
I'm still getting `grpc-bchrpc-node: failed network integrity check` when I switch it to Chronik and my console logs aren't showing up so maybe I need to do a more hard reset of the env?...

**Comment by vorg-code** [comment]
@Klakurka that integrity check happens since forever, we will have to investigate it properly... where did you place the `console.log`?

**Comment by Klakurka** [comment]
All thoughout `chronikservice.ts`.

**Comment by Klakurka** [comment]
I'll leave this up to @chedieck to test properly.

**Comment by vorg-code** [comment]
@Klakurka maybe you forgot to do step 2. 
Also, to see the `console.log` output, after putting it in place in the code, you have to run docker and go into the console of that container, to do so, in the terminal you can type: `docker logs -f paybutton-dev`, then, you can open the sync url in step 3.

Here are both lines you have to alter in the code highlighted and also the output down below of that log in the terminal:
image

**Review by Klakurka — APPROVED** [review]
Needed to recompile.

**Review by chedieck — APPROVED** [review]
Code looks good, test worked.

**Pull request merged**